app/services/image_validation.py

```diff
@@ -0,0 +1,211 @@
+"""
+Image validation service for Voyager-T800.
+
+This module provides image validation functionality to ensure uploaded images
+meet security and quality requirements before processing.
+
+Validation checks:
+- File type: jpg, jpeg, png, webp (by MIME type and magic bytes)
+- File size: maximum 5 MB
+- Resolution: maximum 4096×4096 pixels
+- Image integrity: file can be opened and decoded
+
+All validation errors are logged for debugging and raise HTTPException
+with appropriate status codes and user-friendly messages.
+"""
+
+import io
+import logging
+
+from typing import Dict
+
+from fastapi import HTTPException, UploadFile
+from PIL import Image, UnidentifiedImageError
+
+from app.config.config import settings
+
+
+logger = logging.getLogger(__name__)
+
+
+# Magic bytes for image format detection
+MAGIC_BYTES = {
+    'image/jpeg': [b'\xff\xd8\xff'],
+    'image/png': [b'\x89\x50\x4e\x47\x0d\x0a\x1a\x0a'],
+    'image/webp': [b'RIFF', b'WEBP'],
+}
+
+
+async def validate_image(file: UploadFile) -> Dict[str, any]:
+    """
+    Validate an uploaded image file.
+
+    This function performs validation on an uploaded image:
+    1. Checks file extension
+    2. Checks MIME type (Content-Type header)
+    3. Validates file size
+    4. Verifies magic bytes (actual file content)
+    5. Validates image can be opened and decoded
+    6. Checks resolution limits
+
+    Args:
+        file: FastAPI UploadFile object containing the uploaded image
+
+    Returns:
+        dict: Image metadata including:
+            - filename: Original filename
+            - content_type: MIME type
+            - size_bytes: File size in bytes
+            - width: Image width in pixels
+            - height: Image height in pixels
+
+    Raises:
+        HTTPException:
+            - 400: Validation failed (wrong type, too large, invalid resolution)
+            - 500: Unexpected processing error
+
+    """
+    try:
+        # Step 1: Validate filename and extension
+        filename = file.filename or 'unknown'
```

Comment:
Accepting 'unknown' as default filename may lead to ambiguous error messages. Reject files without a proper filename at the earliest stage.